src/hashmap.c

```diff
@@ -493,3 +493,25 @@ ssize_t hashmap_remove (hashmap_t map, const char *key)
         /* The key was not found, so return 0 */
         return deleted;
 }
+
+/*
+ * Look up the value for a variable.
+ */
+char *lookup_variable (hashmap_t map, const char *varname)
```

Comment:
Of course it looks a clean change, but what is the motivation for this commit? Are you planning to use the function in other modules as well?

Apart from this, the patch does two things:
- it changes the signature to take a hashmap instead of a connprt.
- it moves the function to the hashmap module.
Those changes should probably be two patches.